pkg-manager/hoist/src/index.ts

```diff
@@ -85,7 +87,38 @@ export function getHoistedDependencies (opts: GetHoistedDependenciesOpts) {
 
   const getAliasHoistType = createGetAliasHoistType(opts.publicHoistPattern, opts.privateHoistPattern)
 
-  return hoistGraph(deps, opts.lockfile.importers['.']?.specifiers ?? {}, {
+  const hoistedProjects: Dependency[] = []
+
+  if (opts.hoistWorkspaceProjects) {
+    const allProjects = Object.keys(opts.lockfile.importers)
+
+    const workspaceDir = path.dirname(opts.publicHoistedModulesDir)
+
+    const hoistedWorkspaceProjects =
+        allProjects
+          // Ignore the "." project because it is the root project
+          .filter((projectPath) => projectPath !== '.')
+          .map((projectPath) => {
+            // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
+            const manifest = require(path.join(workspaceDir, projectPath, 'package.json'))
```

Comment:
Instead of reading the `package.json` file again, pass the manifests and workspace project locations from outside. This information is already available in `allProjects` field.

Reply:
Unless I'm missing something the `opts.lockfile.importers` does not have the name and path. The information should be there? https://github.com/pnpm/pnpm/blob/4d34684f1f78aba330c8144b3f3dd93abb7f2da0/lockfile/lockfile-types/src/index.ts#L23-L30

Reply:
you can get it from here: https://github.com/pnpm/pnpm/blob/568860daae87e5077072dd60171ff269e59e0bed/pkg-manager/headless/src/index.ts#L123

and pass it to hoist

Reply:
Oh ok, my bad, I have a constant called `allProjects` too hahaha